src/ray/util/memory.cc

```diff
@@ -27,7 +29,10 @@ uint8_t *pointer_logical_and(const uint8_t *address, uintptr_t bits) {
 
 void parallel_memcopy(uint8_t *dst, const uint8_t *src, int64_t nbytes,
                       uintptr_t block_size, int num_threads) {
-  std::vector<std::thread> threadpool(num_threads);
+  // The compiler generates an atomic operation to ensure the static local variable is
+  // initialized only once even with concurrent callers. Only one thread pool will be
+  // created.
+  static auto &threadpool = *new boost::asio::thread_pool(num_threads);
```

Comment:
num_threads seems wired actually. here when call it twice with different num_threads, it seems not work as expected. @rkooo567 

nit: besides, I think if we create a shared pool, we should put it in global cc file (memory.cc) and not the static member here.

Reply:
Hmm should we just have a static side pool? Thinking more having num_threads as many as hardware limit might be too much (if we have 64 workers, there are going to 64 ** 2 threads context switching for memcpy). Not sure how bad it is.

Reply:
maybe some environment variables should work, for example: RAY_memcp_thread_num

Reply:
@iycheng  what would be the default value then? I believe most of users won't probably specify this.

Reply:
@rkooo567  do you have some recommendations for this?